tools/bazar/services/ExternalBazarService.php

```diff
@@ -4,60 +4,200 @@
 
 use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
 use YesWiki\Wiki;
+use YesWiki\Bazar\Field\ExternalImageField;
 
 class ExternalBazarService
 {
-    public function __construct(Wiki $wiki, ParameterBagInterface $params)
-    {
+    public const FIELD_JSON_FORM_ADDR = 3 ;// replace FIELD_SIZE = 3;
+    public const FIELD_ORIGINAL_TYPE = 4 ;// FIELD_MAX_CHARS = 4;
+
+    private const MAX_CACHE_TIME = 864000 ; // 10 days ot to keep external data in local
+    private const JSON_FORM_BASE_URL = '?BazaR/json&demand=forms&id=';
```

Comment:
passer par l'api?

Reply:
1. l'api n'est pas encore ouverte
2. je préfère garder le /json qui est utilisé partout : rester dans le similaire partout dans le code

Quand on supprimera le /json&demand=forms pour passer à l'api alors oui.

Reply:
en tout cas, je serais partisan de communiquer sur l'api nouvelle version pour la sortie de doryphore et éviter d'utiliser un handler a contre usage.

Mais ok pour garder /json ici, et faire une itération "ouverture de l'api + remplacement de /json"